lib/kml/networklink.ts

```diff
@@ -0,0 +1,232 @@
+import type { KMLOptions } from "lib/kml";
+import { type StyleMap, get1, getMulti, num1 } from "../shared";
+import { extractIconHref, extractStyle } from "./extractStyle";
+import {
+    type Schema,
+    extractCascadedStyle,
+    extractExtendedData,
+    extractTimeSpan,
+    extractTimeStamp,
+    getMaybeHTMLDescription,
+} from "./shared";
+
+interface BoxGeometry {
+    bbox?: BBox;
+    geometry: Polygon;
+}
+
+enum AltitudeMode {
+    ABSOLUTE = "absolute",
+    RELATIVE_TO_GROUND = "relativeToGround",
+    CLAMP_TO_GROUND = "clampToGround",
+    CLAMP_TO_SEAFLOOR = "clampToSeaFloor",
+    RELATIVE_TO_SEAFLOOR = "relativeToSeaFloor",
+}
+
+type BBox = [number, number, number, number];
+
+type LOD = [number, number | null, number | null, number | null];
+interface IRegion {
+    coordinateBox: BoxGeometry | null;
+    lod: LOD | null;
+}
+
+function getNetworkLinkRegion(node: Element): IRegion | null {
+    const region = get1(node, "Region");
+
+    if (region) {
+        return {
+            coordinateBox: getLatLonAltBox(region),
+            lod: getLod(node),
+        };
+    }
+    return null;
+}
+
+function getLod(node: Element): LOD | null {
+    let resLOD: LOD | null = null;
+
+    const lod = get1(node, "Lod");
+
+    if (lod) {
+        resLOD = [
+            num1(lod, "minLodPixels") ?? -1,
+            num1(lod, "maxLodPixels") ?? -1,
+            num1(lod, "minFadeExtent") ?? null,
+            num1(lod, "maxFadeExtent") ?? null,
+        ];
+    }
+
+    return resLOD;
+}
+
+function getLatLonAltBox(node: Element): BoxGeometry | null {
+    const latLonAltBox = get1(node, "LatLonAltBox");
+
+    if (latLonAltBox) {
+        const north = num1(latLonAltBox, "north");
+        const west = num1(latLonAltBox, "west");
+        const east = num1(latLonAltBox, "east");
+        const south = num1(latLonAltBox, "south");
+        const minAlt = num1(latLonAltBox, "minAltitude");
+        const maxAlt = num1(latLonAltBox, "maxAltitude");
+        function processAltitudeMode(mode: Element | null): AltitudeMode | null {
+            switch (mode?.textContent) {
+                case AltitudeMode.ABSOLUTE:
+                    return AltitudeMode.ABSOLUTE;
+                case AltitudeMode.CLAMP_TO_GROUND:
+                    return AltitudeMode.CLAMP_TO_GROUND;
+                case AltitudeMode.CLAMP_TO_SEAFLOOR:
+                    return AltitudeMode.CLAMP_TO_SEAFLOOR;
+                case AltitudeMode.RELATIVE_TO_GROUND:
+                    return AltitudeMode.RELATIVE_TO_GROUND;
+                case AltitudeMode.RELATIVE_TO_SEAFLOOR:
+                    return AltitudeMode.RELATIVE_TO_SEAFLOOR;
+                default:
+                    break;
+            }
+            return null;
+        }
+        const altitudeMode =
+            processAltitudeMode(get1(latLonAltBox, "altitudeMode")) ||
+            processAltitudeMode(get1(latLonAltBox, "gx:altitudeMode"));
```

Comment:
```suggestion
        const altitudeMode =
            processAltitudeMode(get1(latLonAltBox, "altitudeMode") || get1(latLonAltBox, "gx:altitudeMode"));
```

This should get equivalent behavior except in the case where the kml file has both namespaced and non-namespaced elements and the first one has an invalid value, which should be rare enough to be fine as undefined behavior.